src/JSON/Utils/Unicode.dfy

```diff
@@ -0,0 +1,160 @@
+// RUN: %dafny -compile:0 "%s"
```

Comment:
We should be able to use the existing `Unicode` folder here instead - I'll give refactoring a shot and push commits if it works.

Reply:
Never mind, I see the duplication of logic is intentional for performance.

Reply:
To close the loop, I ended up replacing this module entirely with the existing `Unicode` code. See my (forthcoming) summary comment.